polaris/evaluate/utils.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+import pandas as pd
+from typing import Union
+
+from polaris.evaluate import BenchmarkResults, ResultsType
+from polaris.utils.types import PredictionsType
+from polaris.evaluate import Metric
+
+def is_multi_task_single_test_set(vals: PredictionsType, target_cols: list[str]):
+    """Check if the given values are for a multiple-task benchmark with a single
+    test set. This is inferred by comparing the target names with the keys of the
+    given data. If all keys in the given data match the target column names, we
+    assume they are target names (as opposed to test set names for a single-task,
+    multiple test set benchmark)."""
```

Comment:
This makes sense, but I can see this becoming a little messy in the future. Maybe this is something we can think about making more robust after we finish competitions.

Reply:
Yeah I totally agree.. this is at least isolated and explained here now, but we may want to re-think the data format we accept for predictions. There's a [brief discussion about it here](https://valencelabsworkspace.slack.com/archives/C05ASCA5456/p1715960350512859?thread_ts=1715785094.870159&cid=C05ASCA5456). I think it would be reasonable to always expect something like:
```
{"test_set_name": {"target_col_name": [1, 2, 3, 4, 5]}"
```

I get the desire to be as succinct as possible too though.. you could imagine it would be annoying to have to submit that ☝️ every time you just have a list of numbers to submit.